wasmtime-debug/src/gc.rs

```diff
@@ -0,0 +1,232 @@
+use crate::address_transform::AddressTransform;
+use gimli::constants;
+use gimli::read;
+use gimli::{Reader, UnitSectionOffset};
+use std::collections::{HashMap, HashSet};
+use std::vec::Vec;
+
+#[derive(Debug)]
+pub struct Dependencies {
+    edges: HashMap<UnitSectionOffset, HashSet<UnitSectionOffset>>,
+    roots: HashSet<UnitSectionOffset>,
+}
```

Comment:
I'm curious how important doing GC is here. Do we often get debug info with a lot of unreachable content?

Reply:
In most of the cases, the DWARF data is from the linker. The linker just concatenates object files' DWARF sections and updates linked symbol relocation entries. That creates lots of dead entries which could affect memory usage and speed of the DWARF transformation. It is not necessary, but highly advantages for us to purge such records.